### DISCUSSION TITLE
How to turn-off terminal on start?

### DISCUSSION THREAD

**Discussion by krstp** [zed-industries/zed#6581] [Feature Requests]
Currently when Zed launches it brings up terminal pane by default. I don't find it useful. Is there a way to have terminal turned off by default?

**Comment by petros** [comment]
That shouldn't be the case. If you close the terminal tab and quit, next time you open Zed should not have that tab open.

**Reply by SomeoneToIgnore** [reply]
(sorry, I've played with the terminal a bit and the behavior turned out to be more difficult than I've tried to describe in the first version of my reply)

I think, there's still a mismatch with the way Zed opens a text file, being associated as a default program to open it, and the rest of the operations (zed cli, opening via dock) — anything that opens with the latest opened project.

I've had Zed terminal hidden for the default project case and it worked, but for "open the desktop text file" I've always had a terminal open, which I tried to report.
Turns out that if I disable the terminal the same way, next time I open the same desktop file it works and does not bring the terminal tab back.
Until I open the "default project" Zed again — then no matter if I close the default project application and open the desktop file or open the latter straight away — I get the terminal tab opened back despite it being hidden before.

Every other standalone file I open on desktop or elsewhere has its terminal open too, and every separate override gets lost after the default project is reopen.
Not sure that standalone files and open terminal tab is the best defaults, to start with 🙂 

I can make a video, if that helps.

**Reply by krstp** [reply]
Yeah, I think @SomeoneToIgnore got it right… it is a bit wonky, not consistent. I think I will open a feature request. There is also an issue of forgetting opened windows on close. As @SomeoneToIgnore writes there is more to it than meets the eye :)

**Reply by fuzenco** [reply]
Every time I start Zed, terminal opens. I don’t work with projects, just individual files in HTML, CSS, or PHP. I close terminal immediately but it always reopens on startup. It’s the one thing I wish I could turn off.

**Comment by JosephTLyons** [comment]
I'll mention this to the team this week - plenty of people seem to not want this behavior.

**Reply by jbergstroem** [reply]
A configurable setting for the terminal to control this behavior would be welcomed!

**Reply by JosephTLyons** [reply]
I think we will be landing a setting this dev cycle, if the cycle goes according to plan.

**Comment by fuzenco** [comment]
Whoa! It looks like Terminal is off by default now 👍. Thanks…no more having to manually hide it on start-up every time.
